src/volume/element.rs

```diff
@@ -115,34 +208,64 @@ pub trait DataElement:
         E: Endian;
 
     /// Create a single element by converting a scalar value.
-    fn from_u8(value: u8) -> Self;
+    fn from_u8(_value: u8) -> Self {
```

Comment:
Well, many of these cannot practically exist. While a panic is not pretty, in most scenarios you should never get to one of these. Unfortunately rust generics need all of these to exist. We can revisit this if this is a major issue, but I think within the structure of this code as it is, its the only way I can think of using both generics and "covering" impossible conversions.

Reply:
I see that those "implementations" are only needed because  of `RGB8`, `[u8; 3]`, `RGBA8` and `[u8; 4]` which don't call `fn_from_scalar!` or `fn_cplx_from_scalar!` This raises 2 questions:
- What's the difference between those types? To my untrained eyes, `RGB8` and `[u8; 3]` are quite similar. Even their implementations are almost identical.
- Why those 4 types don't need any of the `from_*` functions? Shouldn't they use `from_u8`?

Reply:
RGB8 and [u8; 3] are "under the hood" the same. I added the support for the tuples to allow backwards compatibility to your existing RGB work, which doesn't use a RGB type.
Also Complex types aren't really from scalar, one can make a purely real complex from a scalar but thats it. I doubt thats something that would be needed often in the context of reading or writing a Nifti file.

Reply:
@Enet4 Do you think we should support both `RGB8` and `[u8; 3]`? Idem for `RGBA8` and `[u8; 4]`. I'll be forced to update my code because I'm using `[u8; 3]` but that's not an argument :) It's kind of hard to answer without knowing what other programmers use.

@twitzelbos If we remove `[u8; 3]`, will it still be possible to load my old RGB? I think it will still be possible, but I may misunderstand something.

Reply:
It will still be possible to load, but you won't be able to write it (which you couldn't before anyway it seems).

Reply:
Semantically speaking, `RGB8` and `RGBA8` are pretty accurate for the volume's element type involved, so I would agree with taking these as the _de facto_ way of working with RGB and RGBA volumes.

I do not quite know what it entails to continue supporting both `rgb` structs and the array forms. It would be weird if `Rgb8` and `[u8; 3]` were perceived as different element types (hopefully this is not the case right now). From what I could gather, the binary representation is fully compatible with `[u8; 3]` thanks to `repr(C)`. So internally we can rely mostly on `rgb` and maybe provide zero-cost conversions where suitable (such as being able to read a 2D volume of RGB values into an `Array3<u8>` of dimensions `(H, W, 3)`, or into an `Array2<Rgb8>` of dimensions `(H, W)`).

Reply:
I have a single binary that writes RGB images. None that reads them. I will update my code, no problem. `Array3<RGB8>` does the same thing.

I think it's best if we remove `[u8; 3]` and `[u8; 4]`. It's confusing and it's only useful for backwards compatibility, which we don't care about because we're not at 1.0.0.

Reply:
Ok, will do that. I'll be in the woods with no signal or power until tomorrow night.

Reply:
Done!